cyclops/models/wrappers/sk_model.py

```diff
@@ -141,6 +145,163 @@ def find_best(
 
         return self
 
+    def fit_on_hf_dataset(
+        self,
+        dataset: Dataset,
+        feature_columns: List[str],
+        target_columns: List[str],
+        preprocessor: Optional[ColumnTransformer] = None,
+        batch_size: int = config.DEFAULT_MAX_BATCH_SIZE,
+        **kwargs,
+    ):
+        """Fit the model on a Hugging Face dataset.
+
+        Parameters
+        ----------
+        dataset : Dataset
+             Hugging Face dataset containing features and labels.
+        feature_columns : List[str]
+            List of feature columns in the dataset.
+        target_columns : List[str]
+            List of target columns in the dataset.
+        preprocessor : Optional[ColumnTransformer], optional
+            Transformations to be applied to the data before fitting the model, \
+                by default None
+        batch_size : Optional[int], optional
+            Batch size for batched fitting, used only for estimators with partial fit,
+            by default config.DEFAULT_MAX_BATCH_SIZE
+
+        Returns
+        -------
+        self : `SKModel`
+
+        """
+
+        def fit_model(examples):
+            X_train = np.stack(
+                [examples[feature] for feature in feature_columns], axis=1
+            )
+            if preprocessor is not None:
+                try:
+                    X_train = preprocessor.transform(X_train)
+                except NotFittedError as e:
+                    LOGGER.warn(
+                        f"{e} Fitting preprocessor on batch of size {len(X_train)}"
+                    )
+                    X_train = preprocessor.fit_transform(X_train)
+
+            y_train = np.stack([examples[target] for target in target_columns], axis=1)
+            self.model_.partial_fit(
+                X_train, y_train, classes=np.unique(y_train), **kwargs
+            )
+            return examples
+
+        try:
+            dataset.with_format("numpy", columns=feature_columns + target_columns).map(
+                fit_model,
+                batched=True,
+                batch_size=batch_size,
+            )
+        except AttributeError:
+            LOGGER.info(
+                f"Model {self.model_.__class__.__name__} does not have "
+                "a `partial_fit` method. Calling `fit` directly."
+            )
+            if is_out_of_core(dataset_size=dataset.dataset_size):
+                raise ValueError("Dataset is too large to fit in memory.")
+            ds = dataset.with_format("numpy", columns=feature_columns + target_columns)
```

Comment:
I had missed this. Not exactly sure what this means. you mean the format must not be set necessarily to numpy (or any other format) because it might mess with the transformed data?

Reply:
With the context manager, the format is set temporarily and will be reset to the original format when the context is exited. That could be useful when the dataset is passed by reference or is being returned.